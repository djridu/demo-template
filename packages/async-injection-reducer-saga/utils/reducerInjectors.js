import { combineReducers } from 'redux';
import invariant from 'invariant';
import { isEmpty, isFunction, isString } from 'lodash';

import checkStore from './checkStore';

export function injectReducerFactory(store, isValid) {
    return function injectReducer(key, reducer) {
        if (!isValid) checkStore(store);

        invariant(
            isString(key) && !isEmpty(key) && isFunction(reducer),
            '(src/utils...) injectReducer: Expected `reducer` to be a reducer function',
        );

        if (Reflect.has(store.injectedReducers, key) && store.injectedReducers[key] === reducer)
            return;

        store.injectedReducers[key] = reducer;
        store.replaceReducer(combineReducers({ ...store.injectedReducers }));
    };
}

export default function getInjectors(store) {
    checkStore(store);

    return {
        injectReducer: injectReducerFactory(store, true),
    };
}
