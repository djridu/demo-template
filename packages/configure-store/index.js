// @flow

export { default as configureStore } from './configureStore';
