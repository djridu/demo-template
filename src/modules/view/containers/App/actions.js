/*
 * App Actions
 */

import { LOAD_REPOS, LOAD_REPOS_SUCCESS, LOAD_REPOS_ERROR } from './constants';

export function loadRepos() {
    return {
        type: LOAD_REPOS,
    };
}

export function reposLoaded(repos, username) {
    return {
        type: LOAD_REPOS_SUCCESS,
        payload: {
            repos,
            username,
        },
    };
}

export function repoLoadingError(error) {
    return {
        type: LOAD_REPOS_ERROR,
        payload: {
            error,
        },
    };
}
