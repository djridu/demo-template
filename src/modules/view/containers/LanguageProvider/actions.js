/*
 *
 * LanguageProvider actions
 *
 */

import { CHANGE_LOCALE } from './constants';

export function changeLocale(languageLocale) {
    return {
        type: CHANGE_LOCALE,
        payload: {
            locale: languageLocale,
        },
    };
}
