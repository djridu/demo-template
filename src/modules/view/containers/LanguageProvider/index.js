/*
 *
 * LanguageProvider
 *
 */

import * as React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';
import { IntlProvider } from 'react-intl';

import { makeSelectLocale } from './selectors';

export class LanguageProvider extends React.PureComponent {
    render() {
        return (
            <IntlProvider
                locale={this.props.locale}
                key={this.props.locale}
                messages={this.props.messages[this.props.locale]}
            >
                {React.Children.only(this.props.children)}
            </IntlProvider>
        );
    }
}

LanguageProvider.propTypes = {
    children: PropTypes.element.isRequired,
    locale: PropTypes.string,
    messages: PropTypes.object,
};

const mapStateToProps = createSelector(
    makeSelectLocale(),
    locale => ({
        locale,
    }),
);

export default connect(mapStateToProps)(LanguageProvider);
