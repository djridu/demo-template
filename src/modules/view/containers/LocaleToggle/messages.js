/*
 * LocaleToggle Messages
 */
import { defineMessages } from 'react-intl';

export const scope = 'containers.LocaleToggle';

export default defineMessages({
    en: {
        id: `${scope}.en`,
        defaultMessage: 'en',
    },
    de: {
        id: `${scope}.de`,
        defaultMessage: 'de',
    },
});
