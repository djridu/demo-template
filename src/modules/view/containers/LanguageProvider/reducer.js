/*
 *
 * LanguageProvider reducer
 *
 */

import { createReducer } from 'create-immer-reducer';

import { DEFAULT_LOCALE } from 'utils/i18n';
import { CHANGE_LOCALE } from './constants';

export const initialState = {
    locale: DEFAULT_LOCALE,
};

const languageProviderReducer = createReducer(
    {
        [CHANGE_LOCALE]: (draft, payload) => (draft.locale = payload.locale),
    },
    initialState,
);

export default languageProviderReducer;
