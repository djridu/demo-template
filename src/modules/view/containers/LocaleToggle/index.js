/*
 *
 * LanguageToggle
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import Toggle from 'styled-atoms/Toggle';
import { appLocales } from 'utils/i18n';
import { changeLocale } from 'containers/LanguageProvider/actions';
import { makeSelectLocale } from 'containers/LanguageProvider/selectors';
import Wrapper from './Wrapper';
import messages from './messages';

export class LocaleToggle extends React.PureComponent {
    // eslint-disable-line react/prefer-stateless-function
    render() {
        return (
            <Wrapper>
                <Toggle
                    value={this.props.locale}
                    values={appLocales}
                    messages={messages}
                    onToggle={this.props.onLocaleToggle}
                />
            </Wrapper>
        );
    }
}

LocaleToggle.propTypes = {
    locale: PropTypes.string,
    onLocaleToggle: PropTypes.func,
};

const mapStateToProps = createSelector(
    makeSelectLocale(),
    locale => ({
        locale,
    }),
);

export function mapDispatchToProps(dispatch) {
    return {
        onLocaleToggle: evt => dispatch(changeLocale(evt.target.value)),
        dispatch,
    };
}

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(LocaleToggle);
