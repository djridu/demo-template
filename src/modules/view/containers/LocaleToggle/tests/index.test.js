import * as React from 'react';
import { Provider, ReactReduxContext } from 'react-redux';
import { cleanup, render } from 'react-testing-library';

import { changeLocale } from 'containers/LanguageProvider/actions';
import LanguageProvider from 'containers/LanguageProvider';
import languageProviderReducer from 'containers/LanguageProvider/reducer';
import { configureStore } from 'configure-store';
import { translationMessages } from 'utils/i18n';
import LocaleToggle, { mapDispatchToProps } from '../index';

describe('<LocaleToggle />', () => {
    let store;

    beforeAll(() => {
        const initialState = {
            language: {
                locale: 'EN',
            },
        };
        store = configureStore(initialState, { language: languageProviderReducer }, []);
    });

    afterEach(cleanup);

    it('should match the snapshot', () => {
        const { container } = render(
            <Provider store={store} context={ReactReduxContext}>
                <LanguageProvider messages={translationMessages}>
                    <LocaleToggle />
                </LanguageProvider>
            </Provider>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should present the default `en` english language option', () => {
        const { container } = render(
            <Provider store={store} context={ReactReduxContext}>
                <LanguageProvider messages={translationMessages}>
                    <LocaleToggle />
                </LanguageProvider>
            </Provider>,
        );
        expect(container.querySelector('option[value="en"]')).not.toBeNull();
    });

    describe('mapDispatchToProps', () => {
        describe('onLocaleToggle', () => {
            it('should be injected', () => {
                const dispatch = jest.fn();
                const result = mapDispatchToProps(dispatch);
                expect(result.onLocaleToggle).toBeDefined();
            });

            it('should dispatch changeLocale when called', () => {
                const dispatch = jest.fn();
                const result = mapDispatchToProps(dispatch);
                const locale = 'de';
                const evt = { target: { value: locale } };
                result.onLocaleToggle(evt);
                expect(dispatch).toHaveBeenCalledWith(changeLocale(locale));
            });
        });
    });
});
