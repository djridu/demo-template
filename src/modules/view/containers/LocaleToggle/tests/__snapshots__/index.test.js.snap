// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocaleToggle /> should match the snapshot 1`] = `
<div
  class="Wrapper-sc-1xsw5mc-0 JDLrM"
>
  <select
    class="Select-sc-6vw2nw-0 gcyFDJ"
  >
    <option
      value="en"
    >
      en
    </option>
    <option
      value="de"
    >
      de
    </option>
  </select>
</div>
`;
